python-packages/contract_addresses/setup.py

```diff
@@ -2,17 +2,53 @@
 
 """setuptools module for contract_addresses package."""
 
+# pylint: disable=import-outside-toplevel
+# we import things outside of top-level because 3rd party libs may not yet be
+# installed when you invoke this script
+
 import subprocess  # nosec
-from shutil import rmtree
-from os import environ, path
-from sys import argv
+from shutil import copyfile, rmtree
+from os import environ, path, remove
+from sys import argv, exit  # pylint: disable=redefined-builtin
 
 from distutils.command.clean import clean
 import distutils.command.build_py
 from setuptools import find_packages, setup
 from setuptools.command.test import test as TestCommand
 
 
+class PreInstallCommand(distutils.command.build_py.build_py):
+    """Custom setuptools command class for pulling in addresses.json."""
+
+    description = (
+        "Pull in addresses.json from ../../packages/contract-addresses"
+    )
+
+    def run(self):
+        """Copy over addresses.json."""
+        pkgdir = path.dirname(path.realpath(argv[0]))
+
+        destination_path = path.join(
+            pkgdir, "src", "zero_ex", "contract_addresses"
+        )
+
+        try:
+            remove(path.join(destination_path, "addresses.json"))
+        except FileNotFoundError:
+            pass
```

Comment:
`pass` so that if you run `./setup.py clean` before you've even run `./setup.py pre_install` then it won't unnecessarily throw an execption.  `addresses.json` _is_ mandatory, but during `./setup.py clean` it may or may not be present, and we really don't care if it isn't.

Reply:
Sorry, I was mistaken in my previous response.  I thought we were discussing the `clean` command, not the `pre_install` command.

The idea behind this block was to remove the file before the subsequent call to `copyfile()` puts it in place, in order to avoid an error condition if the file already exists.  However, I just checked the doc for `copyfile` and found that "If dst already exists, it will be replaced," so I went ahead and removed the block in question here, in https://github.com/0xProject/0x-monorepo/pull/2284/commits/01672468166f745712dd41778e3cec4b082fb186